### PR TITLE
Chores: add release-* to monitored branches of Dependabot

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
add release-* to monitored branches of Dependabot

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -16,10 +16,38 @@ updates:
...
-  - package-ecosystem: "npm"
-    directory: "/site"
+  - package-ecosystem: "gomod"
+    directory: "/"
+    schedule:
+      interval: daily
+    labels:
+      - "release-note-none"
+    open-pull-requests-limit: 5
+    target-branch: "release-1.23"
```
Comment (feiskyer) [inline .github/dependabot.yml:26]: is it possible to limit the dependabot only upgrading k8s dependencies for matched versions? (e.g. release-1.23 should only use k8s.io/cloud-provider v0.23.x)

Comment (MartinForReal) [reply .github/dependabot.yml:26]: For now, dependabot doesn't support it. We can ignore k8s related dependencies and manually update them.

Comment (feiskyer) [reply .github/dependabot.yml:26]: yes, looks good

File: .github/dependabot.yml [hunk 2]
```diff
@@ -16,10 +16,38 @@ updates:
       - "release-note-none"
     open-pull-requests-limit: 5
 
-  - package-ecosystem: "npm"
-    directory: "/site"
```
Comment (nilo19) [inline .github/dependabot.yml:20]: why do you want to remove this directory?

Comment (MartinForReal) [reply .github/dependabot.yml:20]: I will revert these changes.

**Comment by MartinForReal** [comment]
related issue: https://github.com/dependabot/dependabot-core/issues/1992

**Comment by MartinForReal** [comment]
@nilo19 @feiskyer Can we bypass ci test pull-cloud-provider-azure-e2e-ccm-capz because this commit is a config change rather than a code change?

**Review by feiskyer — APPROVED** [review]
/lgtm

**Pull request merged**